new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java

```diff
@@ -5,86 +5,175 @@
 
 import java.util.HashMap;
 
-import oracle.weblogic.kubernetes.extensions.LoggedTest;
-
-public class Helm implements LoggedTest {
-
-  private String chartName;
-  private String releaseName;
-  private String namespace;
-  private String repoUrl;
-  private HashMap<String, Object> values;
-
-  public Helm chartName(String chartName) {
-    this.chartName = chartName;
-    return this;
-  }
-
-  public Helm releaseName(String releaseName) {
-    this.releaseName = releaseName;
-    return this;
-  }
-
-  public Helm namespace(String namespace) {
-    this.namespace = namespace;
-    return this;
-  }
-
-  public Helm repoUrl(String repoUrl) {
-    this.repoUrl = repoUrl;
-    return this;
-  }
+import oracle.weblogic.kubernetes.utils.ExecCommand;
+import oracle.weblogic.kubernetes.utils.ExecResult;
+
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class Helm {
+
+  /**
+   * install helm chart
+   * @param params the parameters to helm as values
+   * @return true on success, false otherwise
+   */
+  public static boolean install(HelmParams params) {
+    String namespace = params.getNamespace();
+
+    // assertions for required parameters
+    assertThat(namespace)
+        .as("make sure namespace is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getReleaseName())
+        .as("make sure releaseName is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getRepoUrl() != null && params.getChartName() == null)
+        .as("make sure chart name is not empty or null when repo url is provided")
+        .isFalse();
+
+    assertThat(params.getRepoUrl() == null && params.getChartDir() == null)
+        .as("make sure repo url, chart name and chart dir are not empty or null. "
+            + "repo url, chart name or chart dir must be provided")
+        .isFalse();
+
+    //chart reference to be used in helm install
+    String chartRef = params.getChartDir();
+
+    // use repo url as chart reference if provided
+    if (params.getChartName() != null && params.getRepoUrl() != null) {
+      Helm.addRepo(params.getChartName(), params.getRepoUrl());
+      chartRef = params.getRepoUrl();
+    }
+
+    logger.info(String.format("Installing application in namespace %s using chart %s",
+        namespace, chartRef));
+
+    // build helm install command
+    StringBuffer installCmd = new StringBuffer("helm install ")
+        .append(params.getReleaseName()).append(" ").append(chartRef)
+        .append(" --namespace ").append(params.getNamespace());
 
-  public Helm values(HashMap<String, Object> values) {
-    this.values = values;
-    return this;
-  }
+    // add all the parameters
+    appendValues(params.getValues(), installCmd);
 
-  public String getChartName() {
-    return chartName;
-  }
+    installCmd.append(" && helm list");
```

Comment:
i think this is not ideal - if it fails, how will we know which of the two commands failed?
i think it would be better to run each one separately and then inspect the results
i know is it "more work" - but i think it is much clearer and will produce more reliable code to do if we do it that way

Reply:
will do